packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -459,109 +464,171 @@ export class SDKClient {
     }
   };
 
+  /**
+   * Executes transaction -> handle errors -> capture burned HBAR in metrics and HBAR rate limit class.
+   * @param transaction
+   * @param callerName
+   * @param interactingEntity
+   * @param requestId
+   * @returns
+   */
   private executeTransaction = async (
     transaction: Transaction,
     callerName: string,
     interactingEntity: string,
-    requestId?: string,
-  ): Promise<TransactionResponse> => {
+    requestId: string,
+  ): Promise<TransactionResponse | TransactionResponse[]> => {
     const transactionType = transaction.constructor.name;
-    const requestIdPrefix = formatRequestIdMessage(requestId);
     const currentDateNow = Date.now();
+    let gasUsed: number = 0;
+    let transactionFee: number = 0;
+    let transactionId = [] as string[];
+    let transactionResponse: TransactionResponse | TransactionResponse[] | null = null;
     try {
       // check hbar limit before executing transaction
-      if (this.hbarLimiter.shouldLimit(currentDateNow, SDKClient.recordMode, callerName)) {
+      const shouldLimit = this.hbarLimiter.shouldLimit(currentDateNow, SDKClient.recordMode, callerName);
+      if (shouldLimit) {
         throw predefined.HBAR_RATE_LIMIT_EXCEEDED;
       }
 
       // execute transaction
-      this.logger.info(`${requestIdPrefix} Execute ${transactionType} transaction`);
-      const transactionResponse = await transaction.execute(this.clientMain);
+      // logic: if transaction is typed FileAppendTransaction, use executeAll() to retrieve all fileAppend transaction responses
+      // logic: if transaction is any other type, use execute() to get the only transaction response
+      this.logger.info(`${requestId} Execute ${transactionType} transaction`);
+      if (transactionType === FileAppendTransaction.name) {
+        // execute transaction
+        transactionResponse = await (transaction as FileAppendTransaction).executeAll(this.clientMain);
```

Comment:
Q: Do we know what happens if a subset of FileAppends fail? Does executeAll throw an error?
Something to consider

Reply:
Ah yeah I have thought about this while writing this update. If executeAll() or execute() throw an error, it will fall into the catch block down below. If it fails from executtion level, then `transactionResponse` will be null and at line 582, it'll throw an internal error.

Reply:
Well I mean if there are 3 File Appends and the 3rd one fails on a precheck what happens.
In that case we'd want t be able to capture the fees of the 1st and 2nd FIleAppend

Reply:
Ah that's legit a solid point! 
Yeah that would fall into the catch block in `executeGetTransactionRecord()`, and I don't wanna calculate the transactionFee in the catch block of `executeGetTransactionRecord()` as that'd lead back to the old set up which caused capturing metrics twice. If I add the `addExpense` and `captureMetrics` inside the `for (let txResp of transactionResponse)` loop like Ivo did then it'd be a little messy. Ha interesting it's a diliema! I'll sleep on this a little more then!

Reply:
@Nana-EC new updates are in and at [this line](https://github.com/hashgraph/hedera-json-rpc-relay/pull/2749/files#diff-4b0bcc39d9efb3e923f041c7459f1cc42d34597d4200839df6764c2a345a1c1bR567) and [these lines](https://github.com/hashgraph/hedera-json-rpc-relay/pull/2749/files#diff-4b0bcc39d9efb3e923f041c7459f1cc42d34597d4200839df6764c2a345a1c1bR580-R581), expand/add the `transactionIds` and `transactionFees` of the `FileAppendTx` even in case of an error. This way, the `finally` block will capture all the metrics and log all the `transactionIds` in the `FileAppendTransactions`.